crates/ignore/src/dir.rs

```diff
@@ -290,10 +281,28 @@ impl Ignore {
             errs.maybe_push(err);
             m
         };
+
+        let mut git_dir = dir.join(".git");
```

Comment:
This is introducing an allocation in the common case where previously none existed. Maybe a `Cow<Path>` would help here?

Reply:
reworked to avoid the allocation (and also to cache it if it happens so the net result of this change reduces allocations).